modules/regex/regex.cpp

```diff
@@ -1312,7 +1332,11 @@ Error RegEx::compile(const String& p_pattern) {
 
 						if (min_val != max_val)
 							for (int i = 0; i < stack.size(); ++i)
+#ifdef NO_SAFE_CAST
+								if (stack[i]->is_look_behind())
```

Comment:
I think from a maintenance standpoint it'd be better to use `is_look_behind()` on all builds, taking away the `#ifdef` and the dynamic cast.